### PR TITLE
[Unit Tests] Added unit test for kube proxy builder

### PR DESCRIPTION
**Pull request by srikiz**
Adding unit tests for additional code coverage.
This should include test for nodeup/pkg/model/kube_proxy.go.

I created a new file kube_proxy_test.go that performs few basic checks on how the kube_proxy pod is built.

### CODE REVIEW

**Comment by mikesplain** [comment]
Awesome! Always love more tests, thanks @srikiz!  Lets get this testing!

/ok-to-test

**Comment by srikiz** [comment]
@mikesplain - Thanks, while I wrote this unit test, I didn't actually understand why we need this specific flag for kubeproxy.
_--oom-score-adj=-998_
Any idea where to look this out for?

**Comment by rdrgmnzs** [comment]
@srikiz this is a good explanation of why that flag is set https://github.com/kubernetes/kubernetes/issues/51083#issuecomment-324418659

**Comment by srikiz** [comment]
> @srikiz this is a good explanation of why that flag is set [kubernetes/kubernetes#51083 (comment)](https://github.com/kubernetes/kubernetes/issues/51083#issuecomment-324418659)

Thanks, this sounds good !

**Comment by srikiz** [comment]
@rdrgmnzs @mikesplain - please approve if you are Ok.

File: nodeup/pkg/model/kube_proxy_test.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+			b := &KubeProxyBuilder{
+				NodeupModelContext: tt.fields.NodeupModelContext,
+			}
+			got, err := b.buildPod()
+			if (err != nil) != tt.wantErr {
+				t.Errorf("KubeProxyBuilder.buildPod() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+
+			// compare object metadata Namespace
```
Comment (justinsb) [inline nodeup/pkg/model/kube_proxy_test.go]: What we've done elsewhere is to serialize it to yaml and then string-compare with the expected output.  It tends to make the test a little easier to write, and means we have the output explicitly in the tree which can be pretty handy!

But ... happy to try this approach!

**Comment by justinsb** [comment]
Thanks @srikiz - our yaml "golden output" test approach is also worth considering, but having more test coverage is very welcome - thank you!

/approve
/lgtm

File: nodeup/pkg/model/kube_proxy_test.go (hunk 1 above)
Comment (srikiz) [reply nodeup/pkg/model/kube_proxy_test.go]: sure, thanks @justinsb - I'll keep a note of it for other unit tests.

**Pull request merged**